sox.features.js

```diff
@@ -2022,283 +1982,266 @@
-        });
-
-      const replySpan = $('<span/>', {
-        class: 'newreply added-by-sox',
-        'title': 'link my next chat message as a reply to this',
+      $(document).on('mouseenter', '.mine .message, .mine .timestamp', function() {
+        //Remove excess spacing to the left of the button (by emptying .meta, which has "&nbsp" in it), and set the button color to the background color
+        const metaElement = this.parentElement.querySelector('.meta');
+        metaElement.innerHTML = '';
+        metaElement.style.backgroundColor = sox.helpers.getCssProperty(this.parentElement, 'background-color')
+        metaElement.style.paddingRight = '1px'; // The "padding-right: 1px" is to avoid some weird bug I can't figure out how to fix
+        metaElement.appendChild(replySpan);
+        metaElement.style.display = 'block';
+      }).on('mouseleave', '.mine .message, .mine .timestamp', function() {
+        const metaElement = this.parentElement.querySelector('.meta');
+        metaElement.style.display = 'none';
+        metaElement.querySelector('.newreply').remove();
+      }).on('click', '.newreply.added-by-sox', function() {
+        const message = this.closest('.message');
+        const input = document.querySelector('#input');
+        const rest = input.value.replace(/^:([0-9]+)\s+/, '');
+        input.value = `:${message.id.split('-')[1]} ${rest}`;
+        input.focus();
       });
     },
 
     hideCertainQuestions: function(settings) {
       // Description: Hide certain questions depending on your choices
 
       if (settings.duplicate || settings.closed || settings.migrated || settings.onHold) {
-        $('.question-summary').each(function() { //Find the questions and add their id's and statuses to an object
-          const $question = $(this);
-          const $anchor = $(this).find('.summary a:eq(0)');
-          const text = $anchor.text().trim();
-
-          if (text.substr(text.length - 11) == '[duplicate]' || $question.attr('data-sox-question-state') == 'duplicate') {
-            if (settings.duplicate) $question.hide();
-
-          } else if (text.substr(text.length - 8) == '[closed]' || $question.attr('data-sox-question-state') == 'closed') {
-            if (settings.closed) $question.hide();
-
-          } else if (text.substr(text.length - 10) == '[migrated]' || $question.attr('data-sox-question-state') == 'migrated') {
-            if (settings.migrated) $question.hide();
-
-          } else if (text.substr(text.length - 9) == '[on hold]' || $question.attr('data-sox-question-state') == 'on hold') {
-            if (settings.onHold) $question.hide();
+        [...document.querySelectorAll('.question-summary')].forEach(summary => { // Find the questions and add their id's and statuses to an object
+          const text = summary.querySelectorAll('.summary a')[0].innerText.trim();
+          if ((text.match(/\[duplicate\]$/) && settings.duplicate)
```

Comment:
I think we should use `endsWith` instead of `match` with `$`, regexes seem to be least efficient!